File: 第三章_字符串/字符串(2)/拼接最小字典序练习题/Prior.java
import java.util.*;
 
public class Prior {
    public String findSmallest(String[] strs, int n) {
        if (strs==null||n==0){
            return "";
        }
        Arrays.sort(strs,new Mycomparator());//排序
        StringBuilder builder=new StringBuilder();//字符缓冲器进行字符串拼接
        for (int i=0;i<n;i++){
            builder.append(strs[i]);
        }
        return builder.toString();
    }
 
    private class Mycomparator implements Comparator<String>{//利用比较器进行自定义比较
 
        @Override
        public int compare(String o1, String o2) {
            return (o1+o2).compareTo(o2+o1);
        }
    }
}

File: 第三章_字符串/字符串(2)/拼接最小字典序练习题/Prior.cpp
class Prior {
public:
    string findSmallest(vector<string> strs, int n) {
       int i=n-1;
	   int pos=0;
	   while(i>0){//这里利用了冒泡排序的特点，将数组中的每个元素进行了比较。
		  pos=0;
		  for(int j=0;j<i;j++){
			string str1=strs[j]+strs[j+1];
			string str2=strs[j+1]+strs[j];

			if(str1>str2){
				swap(strs,j,j+1);
				pos=j;
			}
		  }
		  i=pos;
	   }

	   string res;
	   for(i=0;i<n;i++){
		res+=strs[i];
	   }

	   return res;
    }

	void swap(vector<string> &strs,int i,int j)
    {
        string temp=strs[i];
        strs[i]=strs[j];
        strs[j]=temp;
    }
};